jsettlers.logic/src/main/java/jsettlers/logic/movable/components/MultiMaterialComponent.java

```diff
@@ -0,0 +1,58 @@
+package jsettlers.logic.movable.components;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import jsettlers.common.material.EMaterialType;
+
+/**
+ * @author homoroselaps
+ */
+
+public class MultiMaterialComponent extends MaterialComponent {
```

Comment:
Does it make sense to refactor MaterialComponent and MultiMaterialComponent together to a single class?

Reply:
The class `MaterialComponent` implements the function `GetMaterial` of the `ILogicMovable` interface which is used mainly for choosing the correct visuals and for Strategies (like bearer and buildingworker) assuming, that a movable carries only one material at a time. For consistency reasons with the old implementation, I chose `MultiMaterialComponent` to be an extension of the base assumption of "only one material per movable". (although `GetMaterial` becomes ambiguous for more than one material carried)